datagen/src/main/scala/com/nvidia/rapids/tests/scaletest/DataGenEntry.scala

```diff
@@ -0,0 +1,110 @@
+package com.nvidia.rapids.tests.scaletest
+
+import org.apache.spark.sql.SparkSession
+import scopt.OptionParser
+
+object DataGenEntry{
+
+  // case class represents all input commandline arguments
+  private case class Config(scaleFactor: Int = 1,
+    complexity: Int = 1,
+    tables: Array[String] = Array(),
+    outputDir: String = "",
+    format: String = "parquet",
+    version: String = "")
```

Comment:
We need an actual version to put in here. It should not be provided by the user, but should be a way that we can couple the tests with the data being generated. Otherwise the data gen will need a way to know how to generate all previous versions of the data too.

Reply:
It sounds like we need figure out a pattern for the version that conbines the test feature and data feature. I haven't started to create the queries, and also the output data path is named by pattern`SCALE_${scale_factor}_${complexity}_${format}_${version}`, which already contains some basic features. What else information should be added, seed maybe?